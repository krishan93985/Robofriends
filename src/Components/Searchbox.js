import React from 'react';

const searchBox = ({onsearch}) => {
    return (
        <div className='pa2'>
            <input
             className='pa3 ba b--green bg-lightest-blue'
             type='search' 
             placeholder='Search friends'
             onChange={onsearch} 
             />
        </div>
    );
}

export default searchBox;