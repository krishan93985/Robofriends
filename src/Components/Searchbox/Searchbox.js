import React from 'react';

const SearchBox = ({onsearch}) => {
    return (
        <div className='pa2'>
            <input
             aria-label='Search friends'
             className='pa3 ba b--green bg-lightest-blue'
             type='search' 
             placeholder='Search friends'
             onChange={onsearch} 
             />
        </div>
    );
}

export default SearchBox;