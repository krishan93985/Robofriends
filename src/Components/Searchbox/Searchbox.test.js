import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Searchbox from './Searchbox';

let wrapper,inputValueWrapper;

beforeEach(() => {
    wrapper = shallow(<Searchbox />);
})

describe('testing search bar', () => {
    it('search bar renders correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    const event = {
        target:{
            value:'anto'
        }
    }
    const mockSearch = jest.fn();
    inputValueWrapper = shallow(<Searchbox onsearch={mockSearch} />);
    it('changes input value as expected', () => {
        inputValueWrapper.find('input').simulate('change', event);
        expect(mockSearch).toBeCalledWith(event);
    })
})