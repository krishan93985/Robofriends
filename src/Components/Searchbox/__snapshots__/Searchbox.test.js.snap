// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing search bar search bar renders correctly 1`] = `
<div
  className="pa2"
>
  <input
    aria-label="Search friends"
    className="pa3 ba b--green bg-lightest-blue"
    placeholder="Search friends"
    type="search"
  />
</div>
`;
